lmcache/storage_backend/local_backend.py

```diff
@@ -563,3 +564,211 @@ def close(self):
+    """
+
+    def __init__(self,
+                 config: LMCacheEngineConfig,
+                 metadata: LMCacheMemPoolMetadata,
+                 dst_device: str = "cuda"):
+        """
+        Throws:
+            RuntimeError if the loaded configuration does not match the current
+                configuration
+        """
+        super().__init__(config, metadata, dst_device)
+
+        assert config.disk_url is not None, (
+            "Need to specify local path if when "
+            "using LMCLocalDiskBackend")
+        self.proxy = xmlrpc.client.ServerProxy(config.disk_url)
+        # Info: Dict[str, Any] = self.proxy.Info()  # type: ignore
+        # self.remote_chunk_size = Info["chunk_size"]
+
+        self.cached_paths: Dict[CacheEngineKey, str] = {}
+
+        self.written_buffer_size: int = 10
+        self.remain_writing: int = 0
+
+        self.written_key: List[str] = []
+        self.written_size: List[float] = []
+
+        self.update_lock = threading.Lock()
+
+    def contains(
+        self,
+        key: CacheEngineKey,
+    ) -> bool:
+        anws: str = self.proxy.contains(key.to_string)  # type: ignore
+        self.cached_paths[key] = anws
+        return anws != ""
+
+    def batched_contains(self,
+                         keys: Iterable[CacheEngineKey],
+                         total_size: float = 0) -> Iterable[bool]:
+        if total_size == 0:
+            contains_paths: Iterable[str] = self.proxy.batched_contains(
+                [key.to_string() for key in keys])  # type: ignore
+            return [path != "" for path in contains_paths]
+        else:
+            # Write check
+            paths: Iterable[str] = self.proxy.batched_write_check(
+                [key.to_string() for key in keys], total_size)  # type: ignore
+
+            an = []
+            for path, key in zip(paths, keys):
+                if path == "":
+                    an.append(True)
+                else:
+                    self.cached_paths[key] = path
+                    an.append(False)
+            return an
+
+    def save_end_callback(self, future: Future):
+        key, size, chunk_idx = future.result()
+        self.cpu_mbufferpool.idx_free(chunk_idx)
+        # logger.debug(f"Saving cache {key} finished.")
+        self.remain_writing = self.remain_writing - 1
+        self.written_key.append(key.to_string())
+        self.written_size.append(size)
+        if len(self.written_key
+               ) >= self.written_buffer_size or self.remain_writing == 0:
+            self.update_lock.acquire()
+            logger.debug(
+                f"Betched Read Signal for {len(self.written_key)} keys:")
+            # print("Write Ready",self.written_key,self.written_size)
+            signal = self.proxy.batched_write_ready(self.written_key,
+                                                    self.written_size)
+            if not signal:
+                logger.error("Write Ready Signal Failed")
```

Comment:
When does it fail?

Reply:
Only because by default the rpcxml can't return None